pkg/config/viperconfig/viper.go

```diff
@@ -201,8 +201,8 @@ func (c *safeConfig) checkKnownKey(key string) {
 	c.unknownKeys[key] = struct{}{}
 	c.Unlock()
 
-	// log without holding the lock
-	log.Warnf("config key %v is unknown", key)
+	// log without holding the lock. We use stack depth +3 to use the caller function location instead of checkKnownKey
+	log.WarnfStackDepth(3, "config key '%v' is unknown", key)
```

Comment:
Looks like `WarnfStackDepth` needs to be added to the linter to make it happy.

Also, instead of `%v` you could use `%q` to quote the value, since we already know that `key` is a string.